src/wavm.cpp

```diff
@@ -179,7 +171,10 @@ ExecutionResult WavmEngine::execute(
     }
   );
 
-  //delete wavm_host_module::interface; //TODO: clean up, not if there is anybody else using it
+  // clean up
+  WavmEthereumInterface* interface_popped = wavm_host_module::interface.top();
+  wavm_host_module::interface.pop();
+  //delete interface_popped; //TODO how to clean up?
```

Comment:
You could use `std::unique_ptr` to deal with that (or is it `shared_ptr`? @chfast can advise).